src/main/java/com/synopsys/integration/alert/workflow/startup/install/SystemInitializer.java

```diff
@@ -35,27 +39,74 @@
 import com.synopsys.integration.alert.database.provider.blackduck.GlobalBlackDuckConfigEntity;
 import com.synopsys.integration.alert.database.provider.blackduck.GlobalBlackDuckRepository;
 import com.synopsys.integration.alert.database.system.SystemStatusUtility;
+import com.synopsys.integration.alert.workflow.startup.SystemValidator;
 
 @Component
 public class SystemInitializer {
+    private final Logger logger = LoggerFactory.getLogger(SystemInitializer.class);
     private final SystemStatusUtility systemStatusUtility;
     private final AlertProperties alertProperties;
     private final GlobalBlackDuckRepository globalBlackDuckRepository;
     private final EncryptionUtility encryptionUtility;
+    private final SystemValidator systemValidator;
 
     @Autowired
-    public SystemInitializer(final SystemStatusUtility systemStatusUtility, final AlertProperties alertProperties, final GlobalBlackDuckRepository globalBlackDuckRepository, final EncryptionUtility encryptionUtility) {
+    public SystemInitializer(final SystemStatusUtility systemStatusUtility, final AlertProperties alertProperties, final GlobalBlackDuckRepository globalBlackDuckRepository, final EncryptionUtility encryptionUtility,
+        final SystemValidator systemValidator) {
         this.systemStatusUtility = systemStatusUtility;
         this.alertProperties = alertProperties;
         this.globalBlackDuckRepository = globalBlackDuckRepository;
         this.encryptionUtility = encryptionUtility;
+        this.systemValidator = systemValidator;
+    }
+
+    public boolean isSystemInitialized() {
+        return systemStatusUtility.isSystemInitialized();
+    }
+
+    @Transactional
+    public RequiredSystemConfiguration getCurrentSystemSetup() {
+        final Optional<String> proxyHost = alertProperties.getAlertProxyHost();
+        final Optional<String> proxyPort = alertProperties.getAlertProxyPort();
+        final Optional<String> proxyUsername = alertProperties.getAlertProxyUsername();
+        final Optional<String> proxyPassword = alertProperties.getAlertProxyPassword();
+        final Optional<GlobalBlackDuckConfigEntity> blackDuckConfigEntity = getGlobalBlackDuckConfigEntity();
+        String blackDuckUrl = null;
+        Integer blackDuckConnectionTimeout = null;
+        String blackDuckApiToken = null;
+        if (blackDuckConfigEntity.isPresent()) {
+            final GlobalBlackDuckConfigEntity blackDuckEntity = blackDuckConfigEntity.get();
+            blackDuckUrl = blackDuckEntity.getBlackDuckUrl();
+            blackDuckConnectionTimeout = blackDuckEntity.getBlackDuckTimeout();
+            blackDuckApiToken = blackDuckEntity.getBlackDuckApiKey();
+        }
+
+        return new RequiredSystemConfiguration(blackDuckUrl,
```

Comment:
Converting something from an entity to another object seems like a good job for the TypeConverter. If we end up keeping this when multiple providers are added, we should aim for that change.